lib/ansible/modules/cloud/amazon/elasticache_snapshot.py

```diff
@@ -0,0 +1,236 @@
+        changed = True
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Unable to copy the snapshot.", exception=traceback.format_exc(e))
+    return response, changed
+
+def delete(module, connection, name):
+    """ Delete an Elasticache backup. """
+    try:
+        response = connection.delete_snapshot(SnapshotName=name)
+        changed = True
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == "SnapshotNotFoundFault":
+            response = {}
+            changed = False
+        else:
+            module.fail_json(msg="Unable to delete the snapshot.", exception=traceback.format_exc(e))
+    return response, changed
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name            = dict(required=True, type='str'),
+        state           = dict(required=True, type='str', choices=['present', 'absent', 'copy']),
+        replication_id  = dict(required=False, type='str'),
+        cluster_id      = dict(required=False, type='str'),
+        target          = dict(required=False, type='str'),
+        bucket          = dict(required=False, type='str'),
+    )
+    )
+        
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto required for this module')
+
+    name                    = module.params.get('name')
+    state                   = module.params.get('state')
+    replication_id          = module.params.get('replication_id')
+    cluster_id              = module.params.get('cluster_id')
+    target                  = module.params.get('target')
+    bucket                  = module.params.get('bucket')
+
+    # Retrieve any AWS settings from the environment.
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    if not region:
+        module.fail_json(msg = str("Either region or AWS_REGION or EC2_REGION environment variable or boto config aws_region or ec2_region must be set."))
+
+    connection = boto3_conn(module, conn_type='client',
+                            resource='elasticache', region=region,
+                            endpoint=ec2_url, **aws_connect_kwargs)
+
+    changed     = False
+    response    = {}
+
+    if state == 'present':
+        for req in [replication_id, cluster_id]:
+            if not req:
+                module.fail_json(msg="The state 'present' requires options: 'replication_id' and 'cluster_id'")
+        response, changed = create(module, connection, replication_id, cluster_id, name)
+    elif state == 'absent':
+        response, changed = delete(module, connection, name)
+    elif state == 'copy':
+        for req in [target, bucket]:
+            if not req:
+                module.fail_json(msg="The state 'copy' requires options: 'target' and 'bucket'.")
+        response, changed = copy(module, connection, name, target, bucket)
+    
```

Comment:
This line has trailing whitespace, and is causing the shippable PEP8 checks to fail.